src/sdk/pynni/nni/compression/torch/builtin_quantizers.py

```diff
@@ -234,13 +234,47 @@ def __init__(self, model, config_list):
         super().__init__(model, config_list)
 
     def quantize_weight(self, weight, config, **kwargs):
+        weight_bits = get_bits_length(config, 'weight')
         out = weight.tanh()
         out = out / (2 * out.abs().max()) + 0.5
-        out = self.quantize(out, config['q_bits'])
+        out = self.quantize(out, weight_bits)
         out = 2 * out -1
         return out
 
     def quantize(self, input_ri, q_bits):
         scale = pow(2, q_bits)-1
         output = torch.round(input_ri*scale)/scale
         return output
+
+
+class ClipGrad(QuantGrad):
+    @staticmethod
+    def quant_backward(tensor, grad_output, quant_type):
+        if quant_type == 2: # quant_type is quant_output
```

Comment:
`quant_type == 2` --> could you make the quant_type as a enum class? To make readability of the code better

Reply:
fixed.